lib/ui/common/Alert.dart

```diff
@@ -0,0 +1,226 @@
+  final AlertViewModel _viewModel;
+  final AlertStyle _style;
+
+  static present (Alert alert, BuildContext context) {
+    return showDialog<bool>(
+        context: context,
+        barrierDismissible: false,
+        builder: (_) => alert,
+    );
+  }
+
+  const Alert(
+      {Key key,
+      AlertViewModel viewModel,
+      AlertStyle style = _defaultStyle})
+      : this._viewModel = viewModel,
+        this._style = style,
+        super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Material(
+      type: MaterialType.transparency,
+      child: Center(
+        child: Container(
+          margin: _Constants.alertEdgePadding,
+          decoration: BoxDecoration(
+            borderRadius: _Constants.cornerRadius,
+            color: _style.backgroundColor,
+          ),
+          child: Padding(
+            padding: _Constants.alertInnerPadding,
+            child: Column(
+              crossAxisAlignment: CrossAxisAlignment.start,
+              mainAxisSize: MainAxisSize.min,
+              children: <Widget>[
+                Text(
+                  _viewModel.titleText,
+                  style: _style.titleTextStyle,
+                ),
+                SizedBox(
+                  height: _Constants.titleLineSpace,
+                ),
+                Column(
+                  mainAxisSize: MainAxisSize.min,
+                  children: <Widget>[
+                    Text(
+                      _viewModel.detailText,
+                      style: _style.detailTextStyle,
+                    ),
+                    SizedBox(
+                      height: _Constants.detailLineSpace,
+                    ),
+                    Row(
+                      children: _buildAction(context),
+                    ),
+                  ],
+                ),
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  confirmDialog(BuildContext context) {
+    return showDialog<bool>(
+      context: context,
+      barrierDismissible: false,
+      builder: (_) => Alert(),
+    );
+  }
+
+  List<Widget> _buildAction(BuildContext context) {
+    List<Widget> listBuilder = [
+      Expanded(
+        child: RoundedButton(
+          viewModel: RoundedButtonViewModel(
+            title: _viewModel.cancelActionText,
+            onTap: () => Navigator.pop(context),
+          ),
+          style: RoundedButtonStyle.actionSheetLargeRoundedButton().copyWith(
+            height: _Constants.actionHeight,
+            width: _Constants.actionWidth,
+            borderRadius: _Constants.actionCornerRadius,
+          ),
+        ),
+      )
+    ];
+    listBuilder.add(
+      SizedBox(
+        width: _Constants.actionLineSpace,
+      ),
+    );
+    listBuilder.add(Expanded(
+      child: RoundedButton(
+        viewModel: RoundedButtonViewModel(
+            title: _viewModel.confirmActionText,
+            onTap: () => _viewModel.confirmAction(),
```

Comment:
@isaacmartinezimv did you tested if you tap confirm the alert will be dismissed? We want to do both actions the confirmAction from the viewmodel and dismiss of the alert.